src/libraries/Microsoft.Extensions.Hosting/src/HostBuilder.cs

```diff
@@ -182,6 +188,8 @@ internal static DiagnosticListener LogHostBuilding(HostApplicationBuilder hostAp
             return diagnosticListener;
         }
 
+        [UnconditionalSuppressMessage("AOT", "IL3050:RequiresDynamicCode",
```

Comment:
Just double checking that we won't MakeGeneric over the T here?

Reply:
My understanding is that is only a concern for the DiagnosticSource-EventSource bridge in order to write property values out to an EventSource. This isn’t for EventSource listeners. I’m honestly not even sure how that would work because these objects (IHostBuilder, Host) don’t really have properties that can be read. The usage here is that the objects are passed to listeners in the same process in order to get hooks into the building process.

Reply:
I think I've seen that failing in the tests: https://github.com/dotnet/runtime/issues/73420

I've had a bunch of question marks on how this all works, so I've just filed an issue on it and moved on with life.